lionelmssq/fragment_classification.py

```diff
@@ -1,267 +1,102 @@
-            if value - breakage_weight in integer_masses:
-                return True
-    return False
-
-
-def mark_terminal_fragment_candidates(
+def classify_fragments(
     fragment_masses,
     dp_table: DynamicProgrammingTable,
-    ms1_mass=None,
+    breakage_dict: dict,
     output_file_path=None,
-    mass_column_name="neutral_mass",
-    output_mass_column_name="observed_mass",
     intensity_cutoff=0.5e6,
     mass_cutoff=50000,
-    # threshold=MATCHING_THRESHOLD,
-    # ms1_mass_deviations_allowed=0.01,
-):
-    neutral_masses = (
-        fragment_masses.select(pl.col(mass_column_name)).to_series().to_list()
-    )
-
-    integer_masses = [mass.mass for mass in dp_table.masses]
-
-    ms1_compositions = None
-    # if ms1_mass:
-    #     # Compute all compositions for the full sequence (with both tags)
-    #     ms1_compositions = [
-    #         entry
-    #         for entry in explain_mass_with_dp(
-    #             mass=ms1_mass,
-    #             with_memo=True,
-    #             dp_table=dp_table,
-    #             compression_rate=dp_table.compression_per_cell,
-    #             threshold=threshold,
-    #         )
-    #         if entry.breakage == "START_END"
-    #     ][0]
-    #
-    #     if len(ms1_compositions) == 0:
-    #         print("The MS1 mass could not be explained by the nucleosides!")
-
-    # Filter compositions by MS1 mass
-    # TODO: Using the MS1 filter here is not compatible with only doing a table look-up;
-    #  check whether it is still helpful when having composition profiles
-    # for mass in neutral_masses:
-    #     if ms1_compositions:
-    #         compositions = filter_by_ms1_composition(
-    #             fragment_mass=mass, dp_table=dp_table, ms1_mass=ms1_mass,
-    #             ms1_compositions=ms1_compositions,
-    #             threshold=ms1_mass_deviations_allowed
-    #         )
-
-    fragments = fragment_masses.with_columns(
-        pl.Series(neutral_masses).alias(output_mass_column_name)
-    )
-
-    # Determine all fragments that may be terminal ones (start only)
-    fragments = fragments.with_columns(
-        pl.col(mass_column_name)
-        .map_elements(
-            lambda x: is_start_fragment_candidate(x, dp_table=dp_table.table),
-            return_dtype=bool,
+) -> pl.DataFrame:
+    if "observed_mass" in fragment_masses.columns:
+        fragment_masses = fragment_masses.with_columns(
+            pl.lit(intensity_cutoff * 1.1).alias("intensity"),
         )
-        .alias("is_start")
-    )
-
-    # Determine all fragments that may be terminal ones (end only)
-    fragments = fragments.with_columns(
-        pl.col(mass_column_name)
-        .map_elements(
-            lambda x: is_end_fragment_candidate(x, dp_table=dp_table.table),
-            return_dtype=bool,
+    else:
+        fragment_masses = fragment_masses.with_columns(
+            pl.col("neutral_mass").alias("observed_mass"),
```

Comment:
Add a TODO for harmonizing the column names to observed_mass instead of neutral_mass?